### ISSUE TITLE
Why webpage loading is slow when I connect to my shadowsocks server while YouTube videos load fast

### ISSUE THREAD

**Issue by Hollyjolly5** [shadowsocks/shadowsocks-libev#2281]
### What version of shadowsocks-libev are you using?
shadowsocks-libev 3.2.3 (Server)
(Client side: Shadowsocks Android 4.6.5)

### What operating system are you using?
Ubuntu 18.04.1 LTS (Server)
(Client side: Android Pie on Pixel 3 XL)

### What did you do?
First I must say I am a program noob that practically has zero knowledge about coding. But I need to access Google and other websites censored by GFW while I am in China. I read tons of tutorials on the internet and I think [teddysun's guide](https://teddysun.com/486.html) is quite straightforward to follow and requires literally no programming skill.
I followed the guide and deployed a shadowsocks-libev service on a cloud server (Google Cloud Platform or GCP). I chose GCP because it is fast and stable in my area. For most of the time, the ping time to GCP is less than 70 ms, which is considered quiet ideal to many standards. To make my connection speed even improved, I added a traffic control algorithm termed BBR according to another [teddysun's tutorial for newbies](https://teddysun.com/489.html). 

### What did you expect to see?
I had expected a nearly perfect internet experience that circumvents GFW's censorship in a way as if 
the GFW were not in force, given the fact that the communications between my ISP to GCP is almost flawless. 

### What did you see instead?
However, my experience to bypass GFW has been subpar.  
I used an android phone to browse the internet. When I tried to open some webpages, the loading took tens of seconds. But when I opened the YouTube app and played a few videos, it was almost instantaneous. I have no idea what caused the slow loading issue as my knowledge about programming is extremely limited. I had tired my best to record the issue and posted it here so maybe some of your experienced and talented folks can help me out.
Here is what I did.
#### Video demonstration
At local time from 15:15 to 15:18 on Feb 13, I recorded a video showing that how I attempted 3 website loading, which are CNN, Bloomberg and the Verge. None of them loaded at a speed that is satisfactory to me. At the end of the video, I also tried to open the YouTube (App), and it loaded pretty fast. The video can be found [here](https://www.youtube.com/watch?v=-Opk8xj0Pno). 
#### Log file
At the time the video was recording, I pulled Shadowsock-libev's log file to see if there are errors there. I couldn't identify any thing as I literally don't understand what those log entries mean. Please note the server (GCP) time is different from my local time. The server time was from 07:15 to 07:18 when I recorded the video. There are hundreds of entries most of which I think are irrelevant to my issue. Since I don't understand those, I wrote down every single entry from 07:15:33 to 07:17:44 to avoid missing anything that might be useful to your investigation. Log entries after 07:17:44 seem to be just iterations of 
```
Feb 13 07:17:43 instance-2 /usr/local/bin/ss-server[3550]: [udp] server receive a packet
Feb 13 07:17:43 instance-2 /usr/local/bin/ss-server[3550]: [443] [udp] cache hit: 74.125.10.107:443 <-> 203.198.70.29:39679
```
Hence, I did not include any entry after time 07:17:44 in the log file attached. 
Here is the complete log file. 
 [Feb 13 Log.txt](https://github.com/shadowsocks/shadowsocks-libev/files/2859653/Feb.13.Log.txt)

Below are some of the log entries that I think might be useful to you as they contain keywords of CNN, Bloomberg and theverge (three websites I visited in the video demo). Those keyword appeared multiple times in the log file and not all are showed below.
Entries mentioned CNN 
```
Feb 13 07:15:37 instance-2 /usr/local/bin/ss-server[3550]: accept a connection
Feb 13 07:15:37 instance-2 /usr/local/bin/ss-server[3550]: [443] connect to www.cnn.com:443
Feb 13 07:15:37 instance-2 /usr/local/bin/ss-server[3550]: found address name v4 address turner-tls.map.fastly.net
Feb 13 07:15:37 instance-2 /usr/local/bin/ss-server[3550]: found address name v6 address turner-tls.map.fastly.net
Feb 13 07:15:37 instance-2 /usr/local/bin/ss-server[3550]: successfully resolved www.cnn.com
```
Entries mentioned bloomberg
```
Feb 13 07:16:45 instance-2 /usr/local/bin/ss-server[3550]: accept a connection
Feb 13 07:16:45 instance-2 /usr/local/bin/ss-server[3550]: [443] connect to www.bloomberg.com:443
Feb 13 07:16:45 instance-2 /usr/local/bin/ss-server[3550]: found address name v4 address bloomberg.map.fastly.net
Feb 13 07:16:45 instance-2 /usr/local/bin/ss-server[3550]: found address name v6 address bloomberg.map.fastly.net
Feb 13 07:16:45 instance-2 /usr/local/bin/ss-server[3550]: successfully resolved www.bloomberg.com
Feb 13 07:16:45 instance-2 /usr/local/bin/ss-server[3550]: remote connected
Feb 13 07:16:45 instance-2 /usr/local/bin/ss-server[3550]: remote connected
```
Entries mentioned theverge
```
Feb 13 07:17:28 instance-2 /usr/local/bin/ss-server[3550]: [443] connect to www.theverge.com:443
Feb 13 07:17:28 instance-2 /usr/local/bin/ss-server[3550]: [udp] server receive a packet
Feb 13 07:17:28 instance-2 /usr/local/bin/ss-server[3550]: [443] [udp] cache hit: 74.125.204.105:443 <-> 203.198.70.29:38767
Feb 13 07:17:28 instance-2 /usr/local/bin/ss-server[3550]: found address name v4 address vox-chorus.map.fastly.net
Feb 13 07:17:28 instance-2 /usr/local/bin/ss-server[3550]: found address name v6 address vox-chorus.map.fastly.net
Feb 13 07:17:28 instance-2 /usr/local/bin/ss-server[3550]: successfully resolved www.theverge.com
```

### What is your config in detail (with all sensitive info masked)?
```
{   "server":"0.0.0.0",
    "server_port":443,
    "password":"xxxxxxxxxx",
    "timeout":300,
    "user":"nobody",
    "method":"aes-256-cfb",
    "fast_open":true,
    "nameserver":"8.8.8.8",
    "mode":"tcp_and_udp",
    "plugin":"obfs-server",
    "plugin_opts":"obfs=http"
}
```
On the client side, it doesn't matter if the DNS forwarding is toggled on or off. The webpage loading was slow in either case.
Is it a DNS issues ? My limited experience tell me it is unlikely as I think DNS is resolved (forward to 8.8.8.8 ) remotely. I couldn't think of any other causes.

Any help is appreciated. Thank you all.

**Comment by madeye** [comment]
Add `"nameserver": "8.8.8.8",` to your config file and try again.

**Comment by madeye** [comment]
Also try disable `IPv6 routing` in your shadowsocks-android settings.

**Comment by madeye** [comment]
Tried CNN and bloomberg locally, cannot reproduce your issue with both US based and HK based server.

**Comment by madeye** [comment]
Please also disable `fast_open`, it has known perf issues with some ISPs.

**Comment by Hollyjolly5** [comment]
@madeye
1. First I would like to thank you for getting back to me on this issue. I know you are very busy and your guidance and patience are wholeheartedly appreciated. 
I wish I could be more specific about the issue. But as a total amateur, I have tired my best efforts to extract as much information as I could for your guys review (The video demo and the log file). I don't know if you find anything useful there, but at least I hope they are the right directions to have the issue resolved. 
2. Anyhow, I have changed the config file per your instructions. IPv6 route in the shadowsocks android client was never enabled at the first place.
Now the config file looks like this
```
{   "server":"0.0.0.0",
    "nameserver":"1.1.1.1",
    "server_port":443,
    "password":"xxxxxxxxxx",
    "timeout":300,
    "user":"nobody",
    "method":"aes-256-cfb",
    "fast_open":false,
    "nameserver":"8.8.8.8",
    "mode":"tcp_and_udp",
    "plugin":"obfs-server",
    "plugin_opts":"obfs=http"
}
```
Unfortunately, the same issue lingers. Websites loading is slow just like before while YouTube app loads fast as well as its video playback. I know there could be many reasons that caused the issue, but based on your experience is there any other aspect we shall look into in addition to the points you made in your comments above ?

On a separate note, I would like to bring up another thing that I have noticed while I am using an OpenWRT router to browse the internet at home. I think it is relevant to the issue and I think it might be worth to describe it here to provide additional information to you.
The OpenWRT router is basically configured the same as my shadowsock android client. 
[Configuration screenshots](https://imgur.com/a/Aw8WQuE)
Any device that connect to the OpenWRT router would basically have the same experience. YouTube videos load and play almost instantaneously. Google.com and some websites that related to Google (Gmail, Google+, Google Map) load with no or little lag. Third party websites including but not limited to CNN.com, Bloomberg.com theverge.com load at a much slower rate. Sometimes I had to refresh a couple of times to finally get to the site.
When connecting to the shadowsocks-libev service on the GCP, both android client and OpenWRT experienced the same issue (Most websites load very slow). This made me believe the problem is on the server end. 
But again I don't even know where to start to look for the problem on my server. 
Thanks again for reading.

**Comment by madeye** [comment]
It looks any non Google site has poor performance with your server.

Maybe try some other providers.

**Comment by Hollyjolly5** [comment]
I think the issue has something to do with [teddysun's tutorial](https://teddysun.com/486.html) I used to deploy the shadowsock-libev service on my GCP. When I installed a shadowsocks-libev version from [your source](https://github.com/shadowsocks/shadowsocks-libev#install-from-repository), the webpage loading issue seemed gone. Strangely, it reported a version of 3.1.3 after I followed the [install-from-repository](https://github.com/shadowsocks/shadowsocks-libev#install-from-repository) instruction while teddysun's tutorial installed the latest version of 3.2.3 which had been giving me a nuisance. 

My best guess is that some codes in teddysun's script might have caused my issue. However, [his coding](https://github.com/teddysun/shadowsocks_install/blob/master/shadowsocks-all.sh) is beyond my understanding and I cannot pinpoint where might go wrong.
